packages/@react-spectrum/combobox/docs/ComboBox.mdx

```diff
@@ -0,0 +1,767 @@
+        {item => <Item>{item.value.name}</Item>}
+      </ComboBox>
+     </>
+  );
+}
+```
+
+## Complex items
+Items within ComboBox also allow for additional content used to better communicate options. Icons and descriptions can be added to the `children` of `Item` as shown in the example below.
+If a description is added, the prop `slot="description"` must be used to distinguish the different `<Text>` elements.
+
+```tsx example
+<ComboBox label="Select action">
+  <Item textValue="Add to queue">
+    <Add />
+    <Text>Add to queue</Text>
+    <Text slot="description">Add to current watch queue.</Text>
+  </Item>
+  <Item textValue="Add review">
+    <Draw />
+    <Text>Add review</Text>
+    <Text slot="description">Post a review for the episode.</Text>
+  </Item>
+    <Item textValue="Subscribe to series">
+    <Bell />
+    <Text>Subscribe to series</Text>
+    <Text slot="description">Add series to your subscription list and be notified when a new episode airs.</Text>
+  </Item>
+  <Item textValue="Report">
+    <Alert />
+    <Text>Report</Text>
+    <Text slot="description">Report an issue/violation.</Text>
+  </Item>
+</ComboBox>
+```
+
+## Validation
+ComboBox can display a validation state to communicate to the user whether the current value is valid or invalid.
+Implement your own validation logic in your app and pass either `"valid"` or `"invalid"` to the ComboBox via the `validationState` prop.
+
+The example below illustrates how one would validate if the user has entered a valid email into the ComboBox.
+```tsx example
+function Example() {
+  let [value, setValue] = React.useState('me@email.com');
+  let isValid = React.useMemo(() => /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(value), [value]);
+
+  let options = [
+    {id: 1, email: 'fake@email.com'},
+    {id: 2, email: 'anotherfake@email.com'},
+    {id: 3, email: 'bob@email.com'},
+    {id: 4, email: 'joe@email.com'},
+    {id: 5, email: 'yourEmail@email.com'},
+    {id: 6, email: 'valid@email.com'},
+    {id: 7, email: 'spam@email.com'},
+    {id: 8, email: 'newsletter@email.com'},
+    {id: 9, email: 'subscribe@email.com'}
+  ];
+
+  return (
+    <ComboBox
+      width="size-3000"
+      label="To:"
+      validationState={isValid ? 'valid' : 'invalid'}
+      defaultItems={options}
+      value={value}
+      onInputChange={setValue}
+      placeholder="Enter recipient email"
+      allowsCustomValue>
+      {item => <Item>{item.email}</Item>}
+    </ComboBox>
+  );
+}
+```
+
+## Custom Filtering
+By default, ComboBox uses a string "contains" filtering strategy when deciding what items to display in the dropdown menu. This filtering strategy can be overwritten
+by filtering the list of items yourself and passing the filtered list to the ComboBox via the `items` prop.
+
+The example below uses a string "startsWith" filter function obtained from `react-aria`s `useFilter` hook to display items that start with the ComboBox's current input
```

Comment:
I think we'll expose `useFilter` as part of the react spectrum monopackage as we already do for other i18n hooks.

```suggestion
The example below uses a string "startsWith" filter function obtained from the `useFilter` hook to display items that start with the ComboBox's current input
```